py/runtime.c

```diff
@@ -86,7 +86,8 @@ void mp_init(void) {
 
     MP_STATE_VM(mp_reload_exception).base.type = &mp_type_ReloadException;
     MP_STATE_VM(mp_reload_exception).args = (mp_obj_tuple_t *)&mp_const_empty_tuple_obj;
-    MP_STATE_VM(mp_reload_exception).traceback = (mp_obj_traceback_t *)&mp_const_empty_traceback_obj;
+    MP_STATE_VM(mp_reload_exception).traceback = &MP_STATE_VM(mp_reload_traceback);
+    *MP_STATE_VM(mp_reload_exception).traceback = mp_const_empty_traceback_obj;
```

Comment:
Is this right?
```
// Initialize the reload traceback as empty to begin with. It will be filled in later when the exception is raised.
MP_STATE_VM(mp_reload_traceback) = mp_const_empty_traceback_obj;
```

Reply:
> Initialize the reload traceback as empty to begin with. It will be filled in later when the exception is raised.

Not sure what you are asking here exactly but the above statement is correct.

Reply:
I want to suggest the code I put as a clearer version of what you are doing. I was hoping you'd let me know if the comment was correct. This PR doesn't have much explanation of what the original issue was and how you are fixing it.

Reply:
Pointing `traceback` to `const` leads to a crash as `mp_obj_exception_add_traceback` expects a modifiable value.
The crash happens at the following line when trying to re-assign a const struct field:
https://github.com/adafruit/circuitpython/blob/0569c202efbc393dcfc4b95d2472024a6c309b08/py/objexcept.c#L573
Pointing `traceback` to `MP_STATE_VM(mp_reload_traceback)` instead allows a future modification.
Hope this clears things up. :)

Reply:
Since this required a conversation, could you add some comments to the code explicating a bit? Thanks.

Reply:
Sure, I am working on a test for `traceback` module and I'll add this with it.

Reply:
Thank you!